elasticapm/conf/__init__.py

```diff
@@ -572,7 +571,8 @@ class Config(_ConfigBase):
         ],
         default=5,
     )
-    hostname = _ConfigValue("HOSTNAME", default=socket.gethostname())
+    hostname = _ConfigValue("HOSTNAME", default=None)
+    detected_hostname = _ConfigValue("DETECTED_HOSTNAME", default=getfqdn())
```

Comment:
Does this mean that a user can set this via `ELASTIC_APM_DETECTED_HOSTNAME`?

Reply:
Technically, yes. It also allows me to easily override it in tests.